CONTRIBUTING.md

```diff
@@ -28,3 +28,51 @@ Members of the Contributors team are encouraged to review pull requests that
 have already been reviewed, and pull request contributors are encouraged to seek
 multiple reviews.  Reviews from anyone not on the Contributors team are always
 appreciated and encouraged!
+
+## Reviewing Pull Requests
+
+All mlpack contributors who choose to review and provide feedback on Pull
+Requests have a responsibility to both the project and the individual making
+the contribution. 
+
+Reviews and feedback must be
+[helpful, insightful, and geared towards improving the contribution](
+  https://www.youtube.com/watch?v=NNXk_WJzyMI).
+If there are reasons why you feel the PR should not be merged, explain
+what those are. Do not expect to be able to block a Pull Request from advancing
+simply because you say "No" without giving an explanation. Be open to having
+your mind changed. Be open to working with the contributor to make the Pull
+Request better.
+
+Please don't leave dismissive or disrespectful reviews!  It's not helpful for
+anyone.
+
+When reviewing a Pull Request, the primary goals are :
+
+- For the codebase/project to improve
+- For the person submitting the request to succeed
+
+Even if a Pull Request does not gets merged, the submitters should come away
+from the experience feeling like their effort was not wasted or unappreciated.
+Every Pull Request from a new contributor is an opportunity to grow the community. 
+
+When changes are necessary, request them, do not demand them, and do not assume
+that the contributor already knows how to do that. Be there to lend a helping
+hand in case of need.
+
+Since there is clearly a high difference between pull request being raised and
+those being reviewed we highly encourage everyone to review each others pull
+request keeping in mind all the above mentioned points. 
+
+Let's welcome new contributors with ❤️, and not overwhelm them.
+
+## Pull Request Waiting Time 
+
+Since members of the Contributors team only work on mlpack in their free time,
```

Comment:
I guess we could be more open here, because this sounds like only `Contributors` can review PR's, what about: mlpack is a community driven project, it may take some time before a pull request is reviewed.

Reply:
Agreed; if we want to highlight the "free time" bit, maybe `mlpack is a community-driven project, so everyone only works on it in their free time; this means it may take some time ...`

Anyway, just a suggestion, whichever way is fine with me. :+1:

Reply:
Done